src/database/ConcurrencyLog.ts

```diff
@@ -9,8 +10,14 @@ type WriteItem = { type: "write"; tuple: Tuple; txId: TxId | undefined }
 type LogItem = ReadItem | WriteItem
 
 export class ReadWriteConflictError extends Error {
-	constructor(txId: string | undefined) {
-		super("ReadWriteConflictError: " + txId)
+	constructor(txId: string | undefined, writeTuple: Tuple, readBounds: Bounds) {
+		const message = outdent`
+      ReadWriteConflictError: ${txId}
+      Write to tuple ${writeTuple}
+      conflicted with a read at the bounds ${readBounds}
+    `
```

Comment:
Why not `outdent(...)`?

Reply:
🤷🏾‍♂️ that was the author's choice of api: https://www.npmjs.com/package/outdent

Reply:
you would need to do `outdent.string(...)`